src/app/shared/store/registration.state.ts

```diff
@@ -118,20 +124,27 @@ export class RegistrationState {
   CheckAuth({
     patchState,
     dispatch,
-  }: StateContext<RegistrationStateModel>): void {
+  }: StateContext<RegistrationStateModel>): void { 
+    patchState({ isLoading: true });
     this.oidcSecurityService.checkAuth().subscribe((auth) => {
       console.log('is authenticated', auth);
       patchState({ isAuthorized: auth });
-      if (auth) {        
+     
+   
+      if (auth) { 
+        console.log(auth)  
         const token = jwt_decode(this.oidcSecurityService.getToken());
         const id = token['sub'];
         const subrole = token['subrole'];
+       
         this.userService.getUserById(id).subscribe((user) => {
           patchState({ subrole: subrole });
           patchState({ user: user });
+          patchState({ isLoading: false });
```

Comment:
i think this can be moved outside of this "if" block and 147 can be removed

Reply:
No, it can't, we should set isLoading false when response comes. When we will set it earlier, main page present.